cmake/PocoMacros.cmake

```diff
@@ -212,7 +214,14 @@ macro(POCO_MESSAGES out name)
         source_group("${name}\\Message Files" FILES ${ARGN})
         list(APPEND ${out} ${ARGN})
 
-    endif (WIN32)
+    endif (WIN32 AND CMAKE_MC_COMPILER)
+endmacro()
+
+
+macro(POCO_ENABLE_EXE_WMAIN target_name)
+    if(MINGW)
+        set_target_properties("${target_name}" PROPERTIES LINK_FLAGS "-municode")
```

Comment:
`-municode` is needed to compile unicode applications. Should be activated when Uniicode support is enabled in poco or? 
I found also this 

> Help! I get an error: unrecognized command line option "-municode"!

in [MinGW wikiUnicode applications ](https://sourceforge.net/p/mingw-w64/wiki2/Unicode%20apps/)

Normaly CMake takes care that it works on all compilers and platforms. See also CMake rulse:

> 1. Don't make any assumption about the platform or compiler!

I would like to investigate if cmake has any support for such compile flag.

Reply:
Try to compile application with `wmain` using MinGW. You'll get `undefined reference to WinMain`.

Reply:
But wmain is needed when you will have unicode support or? For me it looks like this depends on POCO_ENBALE_WSTRING .... but I'm still investigating...